packages/semcom-sdk/lib/services/query-component-remote.service.ts

```diff
@@ -0,0 +1,26 @@
+import {
+  AbstractQueryComponentService,
+  ComponentMetadata,
+} from '@digita-ai/semcom-core';
+
+export class QueryComponentRemoteService extends AbstractQueryComponentService {
+  private repository: string;
+
+  constructor(repository: string) {
+    super();
+    this.repository = repository;
+  }
+  public query(
+    filter: Partial<ComponentMetadata>,
+  ): Promise<ComponentMetadata[]> {
+    return fetch(`${this.repository}/component`, {
+      method: 'GET',
+      headers: {
+        Accept: 'application/json',
+      },
+      body: JSON.stringify(filter),
+    })
```

Comment:
- This is only the happy flow. What happens if the repository is down/no components were found/...?
 - It isn't standard procedure to add a body to a get-request. It would be better if you make this a `post` request to a new endpoint.